src/main/java/com/google/flourbot/execution/MacroExecutionModuleImplementation.java

```diff
@@ -146,19 +146,33 @@ public ChatResponse execute(String userEmail, String message, String threadId) t
       case SHEET_READ_ROW:
         int row = ((SheetReadRowAction) action).getRow();
```

Comment:
getRow should have all of this special selection logic hidden in it. So you call getRow and you always get a list, even if this happens to be one random field from the list.

Reply:
Great idea! Moving it